### ISSUE TITLE
Add summary of passed/skipped/failed tests to VRL test harness

### ISSUE THREAD

**Issue by bruceg** [timberio/vector#6747]
When running `cargo test`, cargo will print a summary of the numbers of tests run, as well as a listing of which ones failed. The VRL test suite has grown large enough to make scrolling through the listing to see which one(s) failed tedious. This could be improved by providing a similar summary of the test run after completion.

Ref: https://github.com/timberio/vector/pull/6738/checks?check_run_id=2096688778

**Comment by JeanMertz** [comment]
Good idea. Meanwhile, for anyone in need of this, you can use `-f` or `--fail-early` to stop the run on the first error encountered. This won't help with digging through errors on the CI, but at least it helps locally.
